ouroboros-consensus/test-storage/Test/Ouroboros/Storage/ChainDB/TestBlock.hs

```diff
@@ -0,0 +1,288 @@
+import           Ouroboros.Consensus.Node (NodeId (..))
+import           Ouroboros.Consensus.Protocol.Abstract
+import           Ouroboros.Consensus.Protocol.BFT
+
+{-------------------------------------------------------------------------------
+  Test infrastructure: test block
+-------------------------------------------------------------------------------}
+
+newtype TestHash = TestHash Word64
+  deriving newtype (Show, Eq, Ord, Serialise, Num)
+
+
+data TestBlock = TestBlock {
+      tbHash     :: TestHash
+    , tbPrevHash :: ChainHash TestBlock
+    , tbNo       :: Block.BlockNo
+    , tbSlot     :: Block.SlotNo
+    }
+  deriving (Show, Eq)
+
+instance Block.HasHeader TestBlock where
+  type HeaderHash TestBlock = TestHash
+
+  blockHash      = tbHash
+  blockPrevHash  = tbPrevHash
+  blockSlot      = tbSlot
+  blockNo        = tbNo
+  blockInvariant = const True
+
+instance Block.StandardHash TestBlock
+
+instance Measured Block.BlockMeasure TestBlock where
+  measure = Block.blockMeasure
+
+{-------------------------------------------------------------------------------
+  Test infrastructure: ledger state
+-------------------------------------------------------------------------------}
+
+type instance BlockProtocol TestBlock = Bft BftMockCrypto
+
+instance HasPreHeader TestBlock where
+  type PreHeader TestBlock = ()
+  blockPreHeader _ = ()
+
+instance HasPayload (Bft BftMockCrypto) TestBlock where
+  blockPayload _ _ = BftPayload {
+        bftSignature = SignedDSIGN (mockSign () (SignKeyMockDSIGN 0))
+      }
+
+instance UpdateLedger TestBlock where
+  data LedgerState TestBlock =
+      TestLedger {
+          -- The ledger state simply consists of the last applied block
+          lastApplied :: ChainHash TestBlock
+        }
+    deriving (Show)
+
+  data LedgerError TestBlock =
+      -- | The only error possible is that hashes don't line up
+      InvalidHash {
+          expectedHash :: ChainHash TestBlock
+        , invalidHash  :: ChainHash TestBlock
+        }
+    deriving (Show)
+
+  applyLedgerState TestBlock{..} TestLedger{..} =
+      if tbPrevHash == lastApplied
+        then return     $ TestLedger (BlockHash tbHash)
+        else throwError $ InvalidHash lastApplied tbPrevHash
+
+instance ProtocolLedgerView TestBlock where
+  protocolLedgerView _ _ = ()
+
+testInitLedger :: LedgerState TestBlock
+testInitLedger = TestLedger GenesisHash
+
+testInitExtLedger :: ExtLedgerState TestBlock
+testInitExtLedger = ExtLedgerState {
+      ledgerState         = testInitLedger
+    , ouroborosChainState = ()
+    }
+
+-- | Trivial test configuration with a single core node
+testConfig :: NodeConfig (Bft BftMockCrypto)
+testConfig = BftNodeConfig {
+      bftParams   = BftParams { bftSecurityParam = k
+                              , bftNumNodes      = 1
+                              }
+    , bftNodeId   = CoreId 0
+    , bftSignKey  = SignKeyMockDSIGN 0 -- :: SignKeyDSIGN (BftDSIGN c)
```

Comment:
Either remove the comment or comment it _in_?

Reply:
Yup, remnant. Removed.